zk-token-sdk/src/encryption/discrete_log.rs

```diff
@@ -54,30 +59,93 @@ lazy_static::lazy_static! {
             include_bytes!("decode_u32_precomputation_for_G.bincode");
         bincode::deserialize(DECODE_PRECOMPUTATION_FOR_G_BINCODE).unwrap_or_default()
     };
+
+    // pub static ref ITERATOR_G: RistrettoPoint = Scalar::from(NUM_THREADS as u64) * G;
+    // pub static ref THREAD_RANGE_BOUND: usize = (TWO16 as usize) / NUM_THREADS;
 }
 
 /// Solves the discrete log instance using a 16/16 bit offline/online split
 impl DiscreteLog {
+    /// Discrete log instance constructor.
+    ///
+    /// Default number of threads set to 1.
+    pub fn new(generator: RistrettoPoint, target: RistrettoPoint) -> Self {
+        Self {
+            generator,
+            target,
+            num_threads: 1,
+            range_bound: TWO16 as usize,
+            step_point: G,
+        }
+    }
+
+    /// Adjusts number of threads ina  discrete log isntance.
+    pub fn set_number_threads(&mut self, num_threads: usize) -> Result<(), ProofError> {
```

Comment:
very minor nit: rayon uses `num_threads()` for this ([ref](https://docs.rs/rayon/1.5.1/rayon/struct.ThreadPoolBuilder.html#method.num_threads))
```suggestion
    pub fn num_threads(&mut self, num_threads: usize) -> Result<(), ProofError> {
```

Reply:
Yes, in addition, maybe it is cleaner to use the builder pattern here as done in raymon, though there is only one parameter here...

Reply:
I think this is ok as is, no need to reach for the builder pattern in this case IMO

Reply:
Okay, sounds good. I'll update the tests with the new method name and it should be good, thanks! 👍